example/src/main/java/com/stripe/example/view/CustomHorizontalScrollView.java

```diff
@@ -0,0 +1,61 @@
+package com.stripe.example.view;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.widget.HorizontalScrollView;
+
+/**
+ * Created by simonkenny on 05/12/2016.
+ */
+
+public class CustomHorizontalScrollView extends HorizontalScrollView {
+
+    public CustomHorizontalScrollView(Context context) {
+        super(context);
+    }
+
+    public CustomHorizontalScrollView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public CustomHorizontalScrollView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+
+    // lockable behaviour from https://stackoverflow.com/a/5763815
+
+    // true if we can scroll (not locked)
+    // false if we cannot scroll (locked)
+    private boolean mScrollable = true;
+
+    public void setScrollingEnabled(boolean enabled) {
+        mScrollable = enabled;
+    }
+
+    public boolean isScrollable() {
+        return mScrollable;
+    }
+
+    @Override
+    public boolean onTouchEvent(MotionEvent ev) {
+        switch (ev.getAction()) {
+            case MotionEvent.ACTION_DOWN:
+                // if we can scroll pass the event to the superclass
+                if (mScrollable) return super.onTouchEvent(ev);
```

Comment:
Please encase in brackets. I live in perpetual fear of [goto fail](http://www.dwheeler.com/essays/apple-goto-fail.html).